client/src/components/DownloadToken.js

```diff
@@ -39,6 +39,8 @@ export const DownloadToken = () => {
   } = useContext(ScPCAPortalContext)
   const [requesting, setRequesting] = useState(false)
   const [errors, setErrors] = useState([])
+  const buttonLabel =
+    resource && resource.samples ? 'Agree and Continue' : 'Download'
```

Comment:
```suggestion
  const buttonLabel = resource?.samples ? 'Agree and Continue' : 'Download'
```
Alternatively, we can use optional chaining to optimize the check above.